scipy/optimize/_linprog_highs.py

```diff
@@ -47,14 +63,20 @@ def _replace_inf(x):
     return x
 
 
-def _check_invalid_option_values(option, option_str, allowed, default):
-    warning_message = ("Option {0} is {1}, "
-                       "but only values in {2} are allowed. Using default."
-                       .format(option_str, option, allowed))
-    if option not in allowed:
-        warn(warning_message, OptimizeWarning, stacklevel=3)
+def _convert_to_highs_enum(option, option_str, choices, default):
+    # None indicates that we want the default value:
+    if option is None:
+        return default
+    # If option is in the choices we can look it up, if not use
+    # the default value and warn:
+    try:
+        return choices[option.lower()]
+    except (KeyError, AttributeError):
+        warn("Option {0} is {1}, but only values in {2} are allowed."
+             " Using default: {3}".format(
+                 option_str, option, set(choices.keys()), default),
```

Comment:
Might as well convert this to an [f-string](https://realpython.com/python-f-strings/) while we're looking at it.
```suggestion
        warn("Option {option_str} is {option}, but only values in {set(choices.keys())} are allowed."
             " Using default: {default}",
```
Have you looked at the output of this? Don't we want the string corresponding with `default`?

Reply:
Should be changed now. `default` should have the string corresponding to it.  Should be doing the right thing now